app/src/org/commcare/android/database/HybridFileBackedSqlHelpers.java

```diff
@@ -0,0 +1,169 @@
+        // delete files storing data for entries being removed
+        for (Integer id : idsBeingRemoved) {
+            String filename = HybridFileBackedSqlHelpers.getEntryFilename(helper, table, id);
+            if (filename != null && new File(filename).exists()) {
+                files.add(filename);
+            }
+        }
+        return files;
+    }
+
+    protected static void removeFiles(List<String> filesToRemove) {
+        for (String filename : filesToRemove) {
+            File datafile = new File(filename);
+            datafile.delete();
+        }
+    }
+
+    protected static void setFileAsOrphan(SQLiteDatabase db, String filename) {
+        db.beginTransaction();
+        try {
+            ContentValues cv = new ContentValues();
+            cv.put(DatabaseHelper.FILE_COL, filename);
+            db.insertOrThrow(DbUtil.orphanFileTableName, DatabaseHelper.FILE_COL, cv);
+            db.setTransactionSuccessful();
+        } finally {
+            db.endTransaction();
+        }
+    }
+
+    protected static void unsetFileAsOrphan(SQLiteDatabase db, String filename) {
+        int deleteCount = db.delete(DbUtil.orphanFileTableName, DatabaseHelper.FILE_COL + "=?", new String[]{filename});
+        if (deleteCount != 1) {
+            Logger.log(AndroidLogger.SOFT_ASSERT,
+                    "Unable to unset orphaned file: " + deleteCount + " entries effected.b");
+        }
+    }
+
+    public static void removeOrphanedFiles(SQLiteDatabase db) {
+        Cursor cur = db.query(DbUtil.orphanFileTableName, new String[] {DatabaseHelper.FILE_COL}, null, null, null, null, null);
+        ArrayList<String> files = new ArrayList<>();
+        try {
+            if (cur.getCount() > 0) {
+                cur.moveToFirst();
+                int fileColIndex = cur.getColumnIndexOrThrow(DatabaseHelper.FILE_COL);
+                while (!cur.isAfterLast()) {
+                    files.add(cur.getString(fileColIndex));
+                    cur.moveToNext();
+                }
+            }
+        } finally {
+            if (cur != null) {
+                cur.close();
+            }
+        }
+
+        removeFiles(files);
+
+        db.beginTransaction();
+        try {
+            db.delete(DbUtil.orphanFileTableName, null, null);
+            db.setTransactionSuccessful();
+        } finally {
+            db.endTransaction();
+        }
+    }
+
+    protected static File newFileForEntry(File dbDir) throws IOException {
+        File newFile = getUniqueFilename(dbDir);
+
+        if (!newFile.createNewFile()) {
+            throw new RuntimeException("Trying to create a new file using existing filename; " +
+                    "Shouldn't be possible since we already checked for uniqueness");
+        }
+
+        return newFile;
+    }
+
+    private static File getUniqueFilename(File dbDir) {
+        String timeAsString =
+                DateTime.now().toString(DateTimeFormat.forPattern("MM_dd_yyyy_HH_mm_ss"));
```

Comment:
I don't think we need this if we just get a long enough GUID below. Is there a specific reason we want to use the time?